python_modules/dagster/dagster_tests/core_tests/storage_tests/utils/event_log_storage.py

```diff
@@ -1211,6 +1212,88 @@ def _solids():
             )
         ] == [max_record_num, max_record_num - 1]
 
+        # make sure we can accept a run-sharded cursor, where the update_time is some random time
+        # (tzaware)
+        run_cursor_datetime = pendulum.now()
+        assert not list(
+            filter(
+                lambda r: r.storage_id <= 2,
+                storage.get_event_records(
+                    EventRecordsFilter(
+                        after_cursor=RunShardedEventsCursor(
+                            id=2, run_updated_after=run_cursor_datetime
+                        )
+                    )
+                ),
+            )
+        )
+        assert [
+            i.storage_id
+            for i in storage.get_event_records(
+                EventRecordsFilter(
+                    after_cursor=RunShardedEventsCursor(
+                        id=min_record_num + 2, run_updated_after=run_cursor_datetime
+                    ),
+                ),
+                ascending=True,
+                limit=2,
+            )
+        ] == [min_record_num + 3, min_record_num + 4]
+        assert [
+            i.storage_id
+            for i in storage.get_event_records(
+                EventRecordsFilter(
+                    after_cursor=RunShardedEventsCursor(
+                        id=min_record_num + 2,
+                        run_updated_after=run_cursor_datetime,
+                    )
+                ),
+                ascending=False,
+                limit=2,
+            )
+        ] == [max_record_num, max_record_num - 1]
+
+        # make sure we can accept a run-sharded cursor, where the update_time is some random time
+        # (tznaive)
+        run_cursor_naive = pendulum.now().naive()
```

Comment:
could also make this a pytest param since the underlying logic after this var is set appears to be identical?